### DISCUSSION TITLE
Add custom tags with dynamic values for jobs .

### DISCUSSION THREAD

**Discussion by Swejal08** [dagster-io/dagster#20218] [Q&A]
I have a asset_job that runs 3 assets.
```
extract_and_load = define_asset_job(
    name="extract_and_load",
    selection=[
        "download_document",
        "extract_text",
        "load_vector_embedding",
    ],
    hooks={document_processing_failure_hook, document_processing_success_hook},
)

```
 When the first assets runs , I want to add a custom tag with some dynamic value so that i can filter runs in dagster UI based on that tag. 
 Is there any way to achieve this ?
 Also do i need to add some configuration in order to view and filter custom tags in dagster UI ?

**Comment by garethbrickman** [comment]
In Dagster, you can add custom tags to runs, which can then be used for filtering in the Dagster UI. However, tags are typically defined statically when you define the job or when you launch the run.

For your `extract_and_load` job, you can define static tags that will be applied to every run of the job. Here's an example of how you might define a job with static tags:

```python
extract_and_load = define_asset_job(
    name="extract_and_load",
    selection=[
        "download_document",
        "extract_text",
        "load_vector_embedding",
    ],
    tags={"static_key": "static_value"},
    hooks={document_processing_failure_hook, document_processing_success_hook},
)
```

In this example, every run of the `extract_and_load` job will have the tag `{"static_key": "static_value"}`.

If you need to add dynamic tags based on the execution of a specific asset, you would typically need to handle this outside of the asset's execution, such as at the time of launching the run. One way to achieve dynamic tagging could be to use a sensor or a schedule that launches the job with specific tags based on some conditions or states.

Regarding the configuration to view and filter custom tags in the Dagster UI, once tags are associated with runs, they should be visible in the UI without additional configuration. You can filter runs by tags using the search functionality in the UI.

We're requesting input about tags/labels in #19737 
Please feel free to contribute feedback there as well!
